src/Renderer.js

```diff
@@ -39,17 +39,22 @@ export function Renderer(options) {
 
   // use a function, not an arrow, in order to be able to pass "arguments" through
   this.render = function () {
+    const renderArgs = arguments
+    const runRender = () => {
+      if (this._render) {
+        this._render.apply(this, renderArgs)
+      } else {
+        utils.logWarn(`No render function was provided, please use .setRender on the renderer`);
```

Comment:
is this happening frequently? should it throw an event for analytics adapters?

Reply:
I don't know. If the external script is standalone it can handle everything itself but that would be recommended to set a renderer